### ISSUE TITLE
Regarding bev_pool_v2 for asymmery BEV grid

### ISSUE THREAD

**Issue by jiaxiaosong1002** [HuangJunJie2017/BEVDet#128]
Thanks for the open source of the great bev_pool_v2!

Previously, I used the voxel_pooling operation of BEVDepth repo in our dataset with the BEV grid setting:
x_bound=[-14.4, 36.8, 0.399], ## Vehicles in front of the ego vehicle are more important than after.
y_bound=[-25.6, 25.6, 0.4],
z_bound=[-4, 10, 14],
d_bound=[1.0, 41.0, 0.5]
which generates a 128x128 BEV_grid with asymmery x-axis.

I visualize the BEV-feature (no need to train, just random initialization) by:
plt.imshow(np.max(bev_feats[0].cpu().detach().numpy(), axis=0))
![old](https://user-images.githubusercontent.com/81464843/215346017-0ac4216f-24c1-4824-8971-4e6931358112.PNG)
Note that we have rotate the image.

However, when I try to adopt bev_pool_v2, it fails to generate asymmery BEV feature:
![new](https://user-images.githubusercontent.com/81464843/215346167-5b6afe08-273e-4255-aed4-441cf731b01b.PNG)

Note that until the following line in the voxel_pooling_prepare_v2 function:
coor = ((coor - self.grid_lower_bound.to(coor)) / self.grid_interval.to(coor))
The value is exactly the same with BEVDepth, which means the coordinate transformation from camera to ego coordinate system is right.

**I am not sure whether it is bug of the Lift-Split or the acceleration of bev_pool_v2 is based on the assumption that the BEV_grid is symmetric?**

Thanks for your kind help.

A kind reminder @HuangJunJie2017 (Thanks for open source the BEVDet and your blogs on Zhihu are amazing!)

**Comment by HuangJunJie2017** [comment]
@jiaxiaosong1002 
->I am not sure whether it is bug of the Lift-Split or the acceleration of bev_pool_v2 is based on the assumption that the BEV_grid is symmetric?
BEV_grid need not to be symmetric in bev_pool_v2
